### PR TITLE
Improve number input

### PR DESCRIPTION
**Pull request by dkent600**

### CODE REVIEW

File: src/services/numberService.ts [hunk 1]
```diff
@@ -7,13 +7,17 @@ export class NumberService {
    * @param value
    * @param format
    */
-  public toString(value: number, format?: string): string | null {
+  public toString(value: number, format?: string): string | null | undefined {
```
Comment (orenyodfat) [inline src/services/numberService.ts]: is this function is not standard ?

Comment (dkent600) [reply src/services/numberService.ts]: It is a common name, `toString`, but is not everywhere the same API or functionality.

Comment (orenyodfat) [reply src/services/numberService.ts]: why not use standard lib for that ?
what so special in this function that you need to implement that yourself ?
how you guaranty that the call calling this function and not the standard one ?
should it need to be rename  ?

Comment (dkent600) [reply src/services/numberService.ts]: > why not use standard lib for that ?

What `toString` function are you thinking of that matches the API of this one?

> what so special in this function that you need to implement that yourself ?

Primarily, this uses the `numeral` package to convert a number into a string according to the given format.  It also guarantees a string|null|undefined (never NaN).

> how you guaranty that the call calling this function and not the standard one ?

Standard JavaSript class inheritance that we use all the time.  Any method can be safely overridden by any object.  It is not uncommon to override`toString`.

> should it need to be rename?

I don't think there is any need.

Comment (orenyodfat) [reply src/services/numberService.ts]: https://www.w3schools.com/jsref/jsref_tostring_number.asp

Comment (dkent600) [reply src/services/numberService.ts]: That is not the same API, doesn't do anything like what is needed here.

**Pull request merged**